meilisearch/src/extractors/payload.rs

```diff
@@ -59,7 +59,9 @@ impl Stream for Payload {
                         self.limit = new_limit;
```

Comment:
Unfortunately, as you can see here we decrease the limit that we previously stored in the `Payload` struct. What you'll have to do is probably to rename the `limit` field into `current_limit` and decrement it. But also introduce a new limit field that is never modified and that you'll have to use in the error variant (as you have already done).

Reply:
As an example of something that doesn't work:

```bash
cargo run --release -- --http-payload-size-limit 7MiB

# send the movies.json dataset that is 9.5MiB
# https://github.com/ducaale/xh
cat ~/Documents/data/movies.json | xh 'localhost:7700/indexes/products-blable/documents' 'content-type:application/json'
```

As the client is sending the body into multiple blocks, the error message is not correct:

```json
{
    "message": "The provided payload reached the size limit. The maximum accepted payload size is 42.27 KiB.",
    "code": "payload_too_large",
    "type": "invalid_request",
    "link": "https://docs.meilisearch.com/errors#payload_too_large"
}
```

Reply:
Thank you for your feedback and detailed tips on how to improve the code. 

I'd love to help with the test that sends data in blocks but I'll require assistance with that.  

From my observations when the documents are inserted with `curl`/`xh`, the `poll_next` reads the data chunk by chunk, however if test `index.add_documents` is used the stream is consumed immediately. I thought it might be caused by `actix::test` magic, but dunno.
Do you mind explaining me why that happens?

Reply:
It looks indeed more complex than I thought. Adding a test like that will require going quite deep into the `actix::test` libraries. Let's keep your PR that way and merge it 😃